mern/client/src/Contexts/speakerProvider.js

```diff
@@ -0,0 +1,17 @@
+import React from "react";
+
+export const SpeakerContext = React.createContext();
+
+export default function SpeakerProvider({ children }) {
+  // Keeping track of who the speaker is when choosing intent & in transcript
+  const [currSpeaker, setSpeaker] = React.useState("User:");
```

Comment:
Could it be simpler to have a boolean state? Something like isBotSpeaker and just change it accordingly. If it's false, the current speaker is a user. If it's true the current speaker is a bot. The previous speaker is the opposite.

Reply:
Up to preference. Just an idea

Reply:
I thought of this as well but then I feel like it would be extra work to say if it's true, then do "User:" rather than just saying do "User:"

Reply:
Hmm I'm not sure what some advantages using booleans would give us

Reply:
Having a isBot boolean would be better if we used it elsewhere, but since we dont I think what you hve rn is fine 👍